azurerm/resource_arm_container_group.go

```diff
@@ -0,0 +1,317 @@
+				Port: &port,
+			}
+
+			if protocol != "" {
+				containerGroupPort.Protocol = containerinstance.ContainerGroupNetworkProtocol(strings.ToUpper(protocol))
+			}
+
+			containerGroupPorts = append(containerGroupPorts, containerGroupPort)
+		}
+
+		// envVars := data["env_vars"].([]interface{})
+		// if len(envVars) > 0 {
+		// 	envVarsList := make([]containerinstance.EnvironmentVariable, 0, len(envVars))
+
+		// 	for _, envVarRaw := range envVars {
+		// 		envVar := containerinstance.EnvironmentVariable{
+		// 			Name:  &envVarRaw.get("name").(string),
+		// 			Value: &envVarRaw.get("value").(string),
+		// 		}
+		// 		envVarsList = append(envVarsList, envVar)
+		// 	}
+		// 	container.EnvironmentVariables = &envVarsList
+		// }
+
+		containers = append(containers, container)
+	}
+
+	containerGroup := containerinstance.ContainerGroup{
+		Name:     &name,
+		Location: &location,
+		Tags:     expandTags(tags),
+		ContainerGroupProperties: &containerinstance.ContainerGroupProperties{
+			Containers: &containers,
+			IPAddress: &containerinstance.IPAddress{
+				Type:  &IPAddressType,
+				Ports: &containerGroupPorts,
+			},
+			OsType: containerinstance.OperatingSystemTypes(OSType),
+		},
+	}
+
+	_, err := containerGroupsClient.CreateOrUpdate(resGroup, name, containerGroup)
+	if err != nil {
+		return err
+	}
+
+	read, err := containerGroupsClient.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read container group %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmContainerGroupRead(d, meta)
+}
+func resourceArmContainerGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["containerGroups"]
+
+	resp, err := containterGroupsClient.Get(resGroup, name)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	flattenAndSetTags(d, resp.Tags)
+
+	d.Set("os_type", string(resp.OsType))
+	if address := resp.IPAddress; address != nil {
+		d.Set("ip_address_type", address.Type)
+		d.Set("ip_address", address.IP)
+	}
+
+	containerConfigs := flattenContainerInstanceContainers(resp.Containers)
+	d.Set("container", containerConfigs)
+
+	return nil
+}
+
+func resourceArmContainerGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	// container group properties
+	resGroup := id.ResourceGroup
+	name := id.Path["containerGroups"]
+
+	_, err = containterGroupsClient.Delete(resGroup, name)
+
+	if err != nil {
+		return err
```

Comment:
can we check for both a 404 and a dropped connection before throwing an error? we can do this via:

```
resp, err = containterGroupsClient.Delete(resGroup, name)
if err != nil && !utils.ResponseWasNotFound(resp) {
  return nil
}

return nil
```

Reply:
Done.